extension/app/ts/utils/calldata.ts

```diff
@@ -1,107 +1,57 @@
-import { ERC20_APPROVAL_4BYTES, ERC20_TRANSFER_4BYTES, ERC20_TRANSFER_FROM_4BYTES, ERC721_APPROVAL_FOR_ALL_4BYTES } from './constants.js'
-import { decodeMethod } from '@zoltu/ethereum-abi-encoder'
-
-const TRANSFER_PARAMS = [
-	{
-		internalType: 'address',
-		name: 'to',
-		type: 'address'
-	},
-	{
-		internalType: 'uint256',
-		name: 'amount',
-		type: 'uint256'
-	}
-]
-
-const TRANSFER_FROM_PARAMS = [
-	{
-		internalType: 'address',
-		name: 'sender',
-		type: 'address'
-	},
-	{
-		internalType: 'address',
-		name: 'to',
-		type: 'address'
-	},
-	{
-		internalType: 'uint256',
-		name: 'amount',
-		type: 'uint256'
-	}
-]
-
-const APPROVAL_PARAMS = [
-	{
-		internalType: 'address',
-		name: 'spender',
-		type: 'address'
-	},
-	{
-		internalType: 'uint256',
-		name: 'amount',
-		type: 'uint256'
-	},
-]
-
-const APPROVAL_FOR_ALL = [
-	{
-		internalType: 'address',
-		name: 'spender',
-		type: 'address'
-	},
-	{
-		internalType: 'bool',
-		name: 'approved',
-		type: 'bool'
-	}
+import { ethers } from 'ethers'
+import { dataStringWith0xStart } from './bigint.js'
+import * as funtypes from 'funtypes'
+import { EthereumAddress, EthereumQuantity } from './wire-types.js'
+
+const ABI = [
+	'function transfer(address to, uint256 value) public returns (bool success)',
+	'function transferFrom(address from, address to, uint256 value) public returns (bool success)',
+	'function approve(address spender, uint256 value) public returns (bool success)',
+	'function setApprovalForAll(address operator, bool approved)',
 ]
```

Comment:
Consider renaming this.  `abi` usually refers to the solidity output format, while this is just an array of function signatures (not even canonical versions).  This isn't a super well defined thing, but I think it would help avoid confusion by naming this something like `erc20FunctionSignatures`